### PR TITLE
fix: pokedex-bit using CRUD REST

### PR DESCRIPTION
**Pull request by rutujasurve94**
update Jina to 1.1.0

modify flow yamls to enable REST API

add index_rest that indexes docs using the /index REST API

add index_rest to click options

change README to point towards /search (NOT /api/search)

### CODE REVIEW

File: pokedex-with-bit/app.py [hunk 1]
```diff
@@ -3,6 +3,23 @@
...
+import urllib.request
+import gzip
+import numpy as np
+import webbrowser
+import random
+
+from jina.flow import Flow
+from jina import Document
+from jina.clients.sugary_io import _input_files
+from jina.clients.sugary_io import _input_lines
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:22]: clean up

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: pokedex-with-bit/app.py [hunk 2]
```diff
@@ -29,30 +35,94 @@ def index():
...
+        print(f'Indexing {data_path}')
+        url = f'http://0.0.0.0:{f.port_expose}/index'
+
+        input_docs = _input_lines(
+            filepath=data_path,
+            size=num_docs,
+            read_mode='r',
+        )
+        data_json = {'data': [Document(text=text).dict() for text in input_docs]}
+        print(f'#### {len(data_json["data"])}')
```
Comment (cristianmtr) [inline pokedex-with-bit/app.py:60]: remove print

File: pokedex-with-bit/app.py [hunk 3]
```diff
@@ -29,30 +35,94 @@ def index():
...
+            print(
+                f'\n +------------------------------------------------------------------------------------+ \
+                    \n |                                   🤖🤖🤖                                           | \
+                    \n | The directory {workspace} already exists. Please remove it before indexing again.  | \
+                    \n |                                   🤖🤖🤖                                           | \
+                    \n +------------------------------------------------------------------------------------+'
+            )
+            sys.exit(1)
+
+    print(f'### task = {task}')
```
Comment (cristianmtr) [inline pokedex-with-bit/app.py:100]: remove print

**Review by cristianmtr — APPROVED** [review]
Link it to the main ticket, if you haven't

**Review by cristianmtr — APPROVED** [review]

**Review by cristianmtr — DISMISSED** [review]

File: pokedex-with-bit/README.md [hunk 4]
```diff
@@ -74,7 +74,17 @@ After this you should see a new `workspace` folder, which contains all the encod
...
+Follow the Jinabox instructions from the [Query from Docker](#query-from-docker) section above.
+
+### Query With REST API
+```sh
+python app.py -t query_restful
+```
+Then:
+```sh
+curl --request POST -d '{"top_k": 10, "mode": "search",  "data": ["hello world"]}' -H 'Content-Type: application/json' 'http://0.0.0.0:45678/search'
+````
```
Comment (nan-wang) [inline pokedex-with-bit/README.md:86]: Is this part valid?

File: pokedex-with-bit/app.py [hunk 5]
```diff
@@ -29,30 +35,91 @@ def index():
...
+        pass
+
+
+@click.command()
+@click.option('--task', '-t',
+              type=click.Choice(['index', 'query', 'index_restful', 'query_restful', 'dryrun'], case_sensitive=False))
+@click.option('--num_docs_query', '-n', default=100)
+@click.option('--num_docs_index', '-n', default=600)
+def main(task, num_docs_query, num_docs_index):
+    config(task)
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:84]: This leads to the error
```suggestion
    config(task)
```

File: pokedex-with-bit/app.py [hunk 6]
```diff
@@ -29,30 +35,91 @@ def index():
...
-            sys.exit()
-        index()
-    elif sys.argv[1] == 'search':
-        config()
-        search()
+        index(num_docs_index)
+    elif task == 'index_restful':
+        index_restful(num_docs_index)
+    elif task == 'query':
+        config(task)
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:111]: The same here
```suggestion
        config(task)
```

File: pokedex-with-bit/app.py [hunk 7]
```diff
@@ -3,6 +3,11 @@
 
 import os
 import sys
+import click
+import requests
+
+from jina import Document
+from jina.clients.sugary_io import _input_lines
 
 from jina.flow import Flow
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:12]:
```suggestion
```

File: pokedex-with-bit/app.py [hunk 8]
```diff
@@ -3,6 +3,11 @@
 
 import os
 import sys
+import click
+import requests
+
+from jina import Document
```
Comment (nan-wang) [inline pokedex-with-bit/app.py:9]:
```suggestion
from jina import Document, Flow
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: pokedex-with-bit/app.py (hunk 6 above)
Comment (rutujasurve94) [reply pokedex-with-bit/app.py]: Removed code.

File: pokedex-with-bit/app.py (hunk 5 above)
Comment (rutujasurve94) [reply pokedex-with-bit/app.py]: Removed code

File: pokedex-with-bit/app.py (hunk 8 above)
Comment (rutujasurve94) [reply pokedex-with-bit/app.py]: Done

File: pokedex-with-bit/README.md (hunk 4 above)
Comment (rutujasurve94) [reply pokedex-with-bit/README.md]: Removing the `curl` to avoid ambiguity.

**Pull request closed without merging**